// @angular/http/http.ngsummary.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/http';
export function HttpModuleNgSummary() {
    return [{ summaryKind: 2, type: { reference: i0.HttpModule, diDeps: [], lifecycleHooks: [] },
            entryComponents: [], providers: [{ provider: { token: { identifier: { reference: i0.Http } },
                        useClass: null, useValue: undefined, useFactory: { reference: i0.ɵc,
                            diDeps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false,
                                    isOptional: false, token: { identifier: { reference: i0.XHRBackend } } }, { isAttribute: false,
                                    isHost: false, isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.RequestOptions } } }] },
                        useExisting: undefined, deps: [{ isAttribute: false, isHost: false, isSelf: false,
                                isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.XHRBackend } } },
                            { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                token: { identifier: { reference: i0.RequestOptions } } }], multi: false },
                    module: { reference: i0.HttpModule, diDeps: [], lifecycleHooks: [] } },
                { provider: { token: { identifier: { reference: i0.BrowserXhr, diDeps: [],
                                lifecycleHooks: [] } }, useClass: { reference: i0.BrowserXhr, diDeps: [],
                            lifecycleHooks: [] }, useValue: undefined, useFactory: null,
                        useExisting: undefined, deps: [], multi: false }, module: { reference: i0.HttpModule,
                        diDeps: [], lifecycleHooks: [] } }, { provider: { token: { identifier: { reference: i0.RequestOptions } },
                        useClass: { reference: i0.BaseRequestOptions, diDeps: [], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [], multi: false }, module: { reference: i0.HttpModule, diDeps: [],
                        lifecycleHooks: [] } }, { provider: { token: { identifier: { reference: i0.ResponseOptions } },
                        useClass: { reference: i0.BaseResponseOptions, diDeps: [], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [], multi: false }, module: { reference: i0.HttpModule, diDeps: [],
                        lifecycleHooks: [] } }, { provider: { token: { identifier: { reference: i0.XHRBackend,
                                diDeps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false,
                                        isOptional: false, token: { identifier: { reference: i0.BrowserXhr } } }, { isAttribute: false,
                                        isHost: false, isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ResponseOptions } } },
                                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                        token: { identifier: { reference: i0.XSRFStrategy } } }], lifecycleHooks: [] } },
                        useClass: { reference: i0.XHRBackend, diDeps: [{ isAttribute: false, isHost: false,
                                    isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.BrowserXhr } } },
                                { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                    token: { identifier: { reference: i0.ResponseOptions } } }, { isAttribute: false,
                                    isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                    token: { identifier: { reference: i0.XSRFStrategy } } }], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false,
                                isOptional: false, token: { identifier: { reference: i0.BrowserXhr } } }, { isAttribute: false,
                                isHost: false, isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ResponseOptions } } },
                            { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                token: { identifier: { reference: i0.XSRFStrategy } } }], multi: false },
                    module: { reference: i0.HttpModule, diDeps: [], lifecycleHooks: [] } },
                { provider: { token: { identifier: { reference: i0.XSRFStrategy } }, useClass: null,
                        useValue: undefined, useFactory: { reference: i0.ɵb, diDeps: [] },
                        useExisting: undefined, deps: [], multi: false }, module: { reference: i0.HttpModule,
                        diDeps: [], lifecycleHooks: [] } }], modules: [{ reference: i0.HttpModule,
                    diDeps: [], lifecycleHooks: [] }], exportedDirectives: [],
            exportedPipes: [] }, { summaryKind: 3, type: { reference: i0.BrowserXhr, diDeps: [],
                lifecycleHooks: [] } }, { summaryKind: 3, type: { reference: i0.BaseRequestOptions,
                diDeps: [], lifecycleHooks: [] } }, { summaryKind: 3, type: { reference: i0.BaseResponseOptions,
                diDeps: [], lifecycleHooks: [] } }, { summaryKind: 3, type: { reference: i0.XHRBackend,
                diDeps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.BrowserXhr } } }, { isAttribute: false, isHost: false,
                        isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ResponseOptions } } },
                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.XSRFStrategy } } }], lifecycleHooks: [] } }];
}
export function JsonpModuleNgSummary() {
    return [{ summaryKind: 2, type: { reference: i0.JsonpModule, diDeps: [], lifecycleHooks: [] },
            entryComponents: [], providers: [{ provider: { token: { identifier: { reference: i0.Jsonp } },
                        useClass: null, useValue: undefined, useFactory: { reference: i0.ɵd,
                            diDeps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false,
                                    isOptional: false, token: { identifier: { reference: i0.JSONPBackend } } },
                                { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                    token: { identifier: { reference: i0.RequestOptions } } }] }, useExisting: undefined,
                        deps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                token: { identifier: { reference: i0.JSONPBackend } } }, { isAttribute: false, isHost: false,
                                isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.RequestOptions } } }],
                        multi: false }, module: { reference: i0.JsonpModule, diDeps: [], lifecycleHooks: [] } },
                { provider: { token: { identifier: { reference: i0.ɵg, diDeps: [], lifecycleHooks: [] } },
                        useClass: { reference: i0.ɵg, diDeps: [], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [], multi: false }, module: { reference: i0.JsonpModule, diDeps: [],
                        lifecycleHooks: [] } }, { provider: { token: { identifier: { reference: i0.RequestOptions } },
                        useClass: { reference: i0.BaseRequestOptions, diDeps: [], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [], multi: false }, module: { reference: i0.JsonpModule, diDeps: [],
                        lifecycleHooks: [] } }, { provider: { token: { identifier: { reference: i0.ResponseOptions } },
                        useClass: { reference: i0.BaseResponseOptions, diDeps: [], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [], multi: false }, module: { reference: i0.JsonpModule, diDeps: [],
                        lifecycleHooks: [] } }, { provider: { token: { identifier: { reference: i0.JSONPBackend } },
                        useClass: { reference: i0.ɵa, diDeps: [{ isAttribute: false, isHost: false, isSelf: false,
                                    isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ɵg } } },
                                { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                                    token: { identifier: { reference: i0.ResponseOptions } } }], lifecycleHooks: [] },
                        useValue: undefined, useFactory: null, useExisting: undefined,
                        deps: [{ isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false,
                                isOptional: false, token: { identifier: { reference: i0.ɵg } } }, { isAttribute: false,
                                isHost: false, isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ResponseOptions } } }],
                        multi: false }, module: { reference: i0.JsonpModule, diDeps: [], lifecycleHooks: [] } }],
            modules: [{ reference: i0.JsonpModule, diDeps: [], lifecycleHooks: [] }],
            exportedDirectives: [], exportedPipes: [] }, { summaryKind: 3,
            type: { reference: i0.ɵg, diDeps: [], lifecycleHooks: [] } }, { summaryKind: 3,
            type: { reference: i0.BaseRequestOptions, diDeps: [], lifecycleHooks: [] } },
        { summaryKind: 3, type: { reference: i0.BaseResponseOptions, diDeps: [], lifecycleHooks: [] } },
        { summaryKind: 3, type: { reference: i0.ɵa, diDeps: [{ isAttribute: false, isHost: false,
                        isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ɵg } } },
                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.ResponseOptions } } }], lifecycleHooks: [] } }];
}
export function ɵaNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.ɵa, diDeps: [{ isAttribute: false, isHost: false,
                        isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ɵg } } },
                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.ResponseOptions } } }], lifecycleHooks: [] } }];
}
export function BrowserXhrNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.BrowserXhr, diDeps: [], lifecycleHooks: [] } }];
}
export function XHRBackendNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.XHRBackend, diDeps: [{ isAttribute: false,
                        isHost: false, isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.BrowserXhr } } },
                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.ResponseOptions } } }, { isAttribute: false, isHost: false,
                        isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.XSRFStrategy } } }],
                lifecycleHooks: [] } }];
}
export function BaseRequestOptionsNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.BaseRequestOptions, diDeps: [],
                lifecycleHooks: [] } }];
}
export function BaseResponseOptionsNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.BaseResponseOptions, diDeps: [],
                lifecycleHooks: [] } }];
}
export function HttpNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.Http, diDeps: [{ isAttribute: false, isHost: false,
                        isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ConnectionBackend } } },
                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.RequestOptions } } }], lifecycleHooks: [] } }];
}
export function JsonpNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.Jsonp, diDeps: [{ isAttribute: false, isHost: false,
                        isSelf: false, isSkipSelf: false, isOptional: false, token: { identifier: { reference: i0.ConnectionBackend } } },
                    { isAttribute: false, isHost: false, isSelf: false, isSkipSelf: false, isOptional: false,
                        token: { identifier: { reference: i0.RequestOptions } } }], lifecycleHooks: [] } }];
}
export function ɵgNgSummary() {
    return [{ summaryKind: 3, type: { reference: i0.ɵg, diDeps: [], lifecycleHooks: [] } }];
}
